src/Sounder.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2023 Matthew Taylor
+*/
+//
+
+#include "Sounder.h"
+#include "picoPlatform.h"
+#include "KiwiBoardFirmware_menu.h"
+
+PicoPlatform *hardware;
+Beep beepobj[2];
+bool BeepHandler::soundset = 0;
+
+
+void BeepHandler::beep_activate(int tone) {
+
+    int beep = tone;
+    
+    if(soundset) {
+
+        beepobj[beep].beep_activate = 1;
+        Serial.println("Beep activated");
+
+    }
+
+}
+
+// Initialize the end of cycle tone
+void BeepHandler::initTone1() {
+
+    beepobj[0].dot_length=110;
```

Comment:
So, The way this should be done in C++.. 

Your array would normally be an array of Beep instances (change to `Beep * beepobj[2];) , and you would need to construct an instance (`Beep beepOne = new Beep()`).  You then  can set all the fields...  

```
beepOne->dash_length=1000;
....
```

and finally set it to the array 

```
beepobj[1] = beepOne;
```
you would then also change the object method access to be accessed by the arrow operator `->`

To avoid some of this complexity, you might be best served by making `Beep` a struct instead?  It only contains data, and no operations, so it doesn't really need to be a class. Making it a struct would keep you from having to rework all of this